src/main/java/uk/gov/hmcts/reform/civil/service/mediation/helpers/RepresentedLitigantPopulator.java

```diff
@@ -0,0 +1,115 @@
+import uk.gov.hmcts.reform.civil.model.UnavailableDate;
+import uk.gov.hmcts.reform.civil.model.common.Element;
+import uk.gov.hmcts.reform.civil.model.mediation.MediationAvailability;
+import uk.gov.hmcts.reform.civil.model.mediation.MediationContactInformation;
+import uk.gov.hmcts.reform.civil.prd.model.Organisation;
+import uk.gov.hmcts.reform.civil.service.OrganisationService;
+import uk.gov.hmcts.reform.civil.service.mediation.MediationLitigant;
+import uk.gov.hmcts.reform.civil.service.mediation.MediationUnavailability;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+
+import static uk.gov.hmcts.reform.civil.enums.YesOrNo.YES;
+import static uk.gov.hmcts.reform.civil.enums.dq.UnavailableDateType.DATE_RANGE;
+import static uk.gov.hmcts.reform.civil.enums.dq.UnavailableDateType.SINGLE_DATE;
+import static uk.gov.hmcts.reform.civil.utils.ElementUtils.unwrapElements;
+
+@Component
+public class RepresentedLitigantPopulator {
+
+    private final OrganisationService organisationService;
+
+    private static final String DATE_FORMAT = "yyyy-MM-dd";
+
+    public RepresentedLitigantPopulator(OrganisationService organisationService) {
+        this.organisationService = organisationService;
+    }
+
+    public MediationLitigant.MediationLitigantBuilder populator(MediationLitigant.MediationLitigantBuilder builder,
+                                                                MediationContactInformation mediationContactInformation,
+                                                                MediationAvailability mediationAvailability,
+                                                                OrganisationPolicy organisationPolicy, String solicitorEmail) {
+
+        String solicitorOrgName = getSolicitorOrgName(organisationPolicy);
+        String mediationContactName = getMediationContactName(mediationContactInformation);
+        String mediationContactNumber = mediationContactInformation != null
+            ? mediationContactInformation.getTelephoneNumber() : null;
+        String mediationEmail = mediationContactInformation != null
+            ? mediationContactInformation.getEmailAddress() : null;
+
+        List<MediationUnavailability> dateRangeToAvoid = getDateRangeToAvoid(mediationAvailability);
+
+        return builder.represented(true)
+            .solicitorOrgName(solicitorOrgName)
+            .litigantEmail(solicitorEmail)
+            .litigantTelephone(null)
+            .mediationContactName(mediationContactName)
+            .mediationContactNumber(mediationContactNumber)
+            .mediationContactEmail(mediationEmail)
+            .dateRangeToAvoid(dateRangeToAvoid);
+    }
+
+    @Nullable
+    private String getSolicitorOrgName(OrganisationPolicy organisationPolicy) {
+        String orgId = organisationPolicy.getOrganisation().getOrganisationID();
```

Comment:
please add null check - i have seen in prod

Cannot invoke "[uk.gov](http://uk.gov/).hmcts.reform.ccd.model.Organisation.getOrganisationID()" because the return value of "[uk.gov](http://uk.gov/).hmcts.reform.ccd.model.OrganisationPolicy.getOrganisation()" is null